app/policies/collection_policy.rb

```diff
@@ -30,6 +30,37 @@ def authorize!(action, message = nil)
     raise NotAuthorizedError, message unless public_send(action)
   end
 
+  protected
+
+  def authority
+    Services.checkpoint
+  end
+
+  def can?(action,resource)
+    Checkpoint::Query::ActionPermitted.new(user, action, resource, authority: authority).true?
+  end
+
+  def all_of_type(type)
+    resource = Checkpoint::Resource::AllOfType.new(type)
+  end
+
+  def scope_for_resource(resource)
+    if resource.all?
+      scope.all
+    elsif resource.all_of_type?
+      scope.with_type(resource.type)
+    else
+      scope.with_type_and_id(resource.type,resource.id)
+    end
+  end
+
+  def showable_scopes
```

Comment:
Agreed the block is probably not the right thing here, just an expedient way to pull up the notion of the scope that contains all things for which a user can do a particular action.

Another possibility would be to have e.g. `showable` be a scope on the model (passing the authority as a parameter) - Right now I think our dividing line is that although the model knows how to convert itself to a resource token, the policy is essentially responsible for the inverse (mapping from a resource token to a scope) - I think there could be an argument for moving the basically all but the `authority.which(user, :show)` to the model.

Reply:
But, let's try `resource_types` for now and see how that feels.

Reply:
I think once `showable_scope` calls `resource_types`, then `showable_scope` just becomes the default `resolve` method.

Reply:
I sort of had that inkling, too. I'm curious to see if we find any need to distinguish -- I suspect not, because this is the part where cancan and pundits implementations got the ugliest: "X's for y action".

I think those would fundamentally be operation policies, rather than collection or resource policies. An alternative would be operation-specific subclasses of the collection policy, though I haven't thought that through yet. In any case, combining those multi-modal responsibilities complicates the policy (in isolation, at best) and potentially the model (by leaking policy decisions into scopes). I think we do well in maintaining a strict (policy class) division between "for any purpose" access from "for some given purpose".

Reply:
The latest commits use `resource_types` and a default `resolve` implementation, but having done away with the non-Package policies I wonder if the fancy resolve should get pushed back down to `PackagePolicy`, and `Audit` maybe doesn't need to support `resource_type`.

Reply:
Well, I guess `Audit` still needs to support figuring out which audits the user can view -- right now, normally none unless they're an admin, but that's based on understanding what grants we intend to create - it would certainly be possible and sensible to grant a user admin but restricted to audits rather than the entire repository. So, I think it makes sense to keep the fancy `resolve` at the `CollectionPolicy` level.